### PR TITLE
issue/3041 Removed view management

### LINKED ISSUE #3041: adapt-contrib-bookmarking: view management
### Subject of the issue
Bookmarking currently captures all rendered views but it does not release them if they are removed by another extension (such as branching).
It doesn't need to capture views.

### Aim
Remove dependency on view references.

### PR DESCRIPTION
**Pull request by oliverfoster**
fixes https://github.com/adaptlearning/adapt_framework/issues/3041

### Changes
* Removed view management, reverted to live element discovery

### CODE REVIEW

**Comment by moloko** [comment]
@oliverfoster I'm not sure why but when running this version I don't get any bookmarking after the first component it bookmarks on the page. Looks like `possibleViewIds` keeps coming up empty.

**Review by cahirodoherty-learningpool — APPROVED** [review]
👀

**Comment by oliverfoster** [comment]
> @oliverfoster I'm not sure why but when running this version I don't get any bookmarking after the first component it bookmarks on the page. Looks like `possibleViewIds` keeps coming up empty.

It's probably because I misunderstood `currentLocationId`

File: js/adapt-contrib-bookmarking.js [hunk 1]
```diff
@@ -6,6 +6,7 @@ class Bookmarking extends Backbone.Controller {
     this.bookmarkLevel = null;
     this.restoredLocationID = null;
     this.currentLocationID = null;
+    this.currentContentObjectId = null;
```
Comment (moloko) [inline js/adapt-contrib-bookmarking.js]: Could use `Adapt.location._currentId` as that holds the same data?

**Review by moloko — APPROVED** [review]
one suggestion, otherwise looks good (tested in FF and IE11 using both component and block level bookmarking)

File: js/adapt-contrib-bookmarking.js (hunk 1 above)
Comment (oliverfoster) [reply js/adapt-contrib-bookmarking.js]: Genius

**Review by moloko — APPROVED** [review]
lovely stuff

File: js/adapt-contrib-bookmarking.js [hunk 2]
```diff
@@ -177,52 +175,39 @@ class Bookmarking extends Backbone.Controller {
...
   releaseViews() {
-    this.watchViews.length = 0;
-    this.watchViewIds.length = 0;
     this.stopListening(Adapt, 'remove', this.releaseViews);
-    this.stopListening(Adapt, this.bookmarkLevel + 'View:postRender', this.captureViews);
     $(window).off('scroll', this._onScroll);
   }
 
   checkLocation() {
+    if (!this.currentLocationID) return;
```
Comment (moloko) [inline js/adapt-contrib-bookmarking.js]: one last thing: just wondering why we need this check..?

Comment (oliverfoster) [reply js/adapt-contrib-bookmarking.js]: we don't really

**Review by moloko — APPROVED** [review]
perfection

**Review by guywillis — APPROVED** [review]
🚀

**Review by cahirodoherty-learningpool — APPROVED** [review]
👀

**Pull request merged**